src/IR/IR.ts

```diff
@@ -64,6 +73,20 @@ export interface Assignment {
   expr: Expr;
 }
 
+export type VariableType =
```

Comment:
VariableType seems to be missing some types (list<?>, table<?,?>). Those are more complicated to represent; maybe we just leave off the types for now; see next comment.